### PR TITLE
feat: publish the solution to GitHub Pages

### LINKED ISSUE #4: Publish the solution to GitHub Pages
This will let users see what they are going to develop.
**Use github actions** to publish to github pages https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/ and **not a branch**. We don't need to keep history of the application and the GH actions are much more flexible than publishing using branches (in particular, Jekkyl is not involved).

Tasks
- [x] Setup automation: publish when the code changes in the `solution` branch
- [x] Add link to the github pages environment in the repo description
- [x] Add link in the README of the `develop` branch
- [x] Add link in the README of the `solution` branch

### PR DESCRIPTION
**Pull request by csouchet**
I used the [Vite](https://vitejs.dev/guide/static-deploy.html#github-pages) documentation to deploy on GitHub Pages and the GitHub actions:
- https://github.com/marketplace/actions/deploy-github-pages-site
- https://github.com/marketplace/actions/upload-github-pages-artifact
- https://github.com/actions/starter-workflows/tree/main/pages

Covers #4

### CODE REVIEW

File: .github/workflows/deploy-solution.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Deploy the solution
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push event but only for the "solution" branch
+  push:
+    branches: [ "solution" ]
```
Comment (tbouffard) [inline .github/workflows/deploy-solution.yml:7]: can we temporary trigger the work when pushing on the branch of this PR to test that the deployment can work?
This also means changing the configuration temporary and also fix it to at the end only authorize publish from the solution branch.

image

**Review by tbouffard — COMMENTED** [review]
LGTM, I suggest we test the deployment before merging

File: .github/workflows/deploy-solution.yml (hunk 1 above)
Comment (csouchet) [reply .github/workflows/deploy-solution.yml:7]: image

The [page](https://process-analytics.github.io/bpmn-visualization-tutorial-getting-started/) is empty 😭

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "license": "Apache-2.0",
     "scripts": {
       "start": "vite",
-      "build": "vite build",
+      "build": "vite build --base=/bpmn-visualization-tutorial-getting-started/",
```
Comment (tbouffard) [inline package.json:8]: ⚠️ The `--base` option prevents to test locally.
I suggest we also add it to the preview script (I have tested it locally) and that we document in the README how to test the production build locally.
@csouchet I can handle it if you agree with the proposal

**Comment by tbouffard** [comment]
ℹ️ Current configuration. The branch of this PR should be removed from the configuration once this PR is merged

image

File: package.json (hunk 2 above)
Comment (csouchet) [reply package.json:8]: Ok, for me 🙂

Comment (tbouffard) [reply package.json:8]: done in 63e67f05

**Review by tbouffard — APPROVED** [review]
LGTM

**Comment by tbouffard** [comment]
Configuration updated

image

**Pull request merged**